proton

```diff
@@ -165,9 +168,10 @@ class CompatData:
                         dirs.append(path)
             for d in dirs:
                 try:
-                    os.rmdir(d)
-                except OSError:
-                    #not empty
+                    if not any(os.scandir(d)):
```

Comment:
This introduces a race condition. Please rather change OSError to FileNotFoundError since this is Python 3